src/pycram/helper.py

```diff
@@ -22,6 +23,13 @@
 import os
 import math
 
+class Singleton(type):
+    _instances = {}
```

Comment:
Also here docstring

Reply:
the instances dict still needs and explanation